src/PhpSpec/Formatter/TapFormatter.php

```diff
@@ -0,0 +1,161 @@
+
+    const PLAN = '1..%d';
+
+    const MESSAGE = "---\nmessage: %s\n...";
+
+    const SEVERITY = "\n---\nseverity: %s\n...";
+
+    /**
+     * @var int
+     */
+    private $examplesCount = 0;
+
+    /**
+     * @var string
+     */
+    private $currentSpecificationTitle;
+
+    /**
+     * @param SuiteEvent $event
+     */
+    public function beforeSuite(SuiteEvent $event)
+    {
+        $this->getIO()->writeln(self::VERSION);
+    }
+
+    /**
+     * @param SpecificationEvent $event
+     */
+    public function beforeSpecification(SpecificationEvent $event)
+    {
+        $this->currentSpecificationTitle = $event->getSpecification()->getTitle();
+    }
+
+    /**
+     * @param ExampleEvent $event
+     */
+    public function afterExample(ExampleEvent $event)
+    {
+        $this->examplesCount++;
+        $desc = sprintf(
+            self::DESC,
+            $this->currentSpecificationTitle,
+            preg_replace('/^its? /', '', $event->getExample()->getTitle())
+        );
+
+        switch ($event->getResult()) {
+            case ExampleEvent::PASSED:
+                $result = sprintf(self::OK, $this->examplesCount) . $desc;
+                break;
+            case ExampleEvent::PENDING:
+                $message = $this->getResultData($event, $event->getResult());
+                $result = sprintf(self::NOT_OK, $this->examplesCount) . $desc . $message;
+                break;
+            case ExampleEvent::SKIPPED:
+                $message = sprintf(self::SKIP, $this->getResultData($event));
+                $result = sprintf(self::OK, $this->examplesCount) . $desc . $message;
+                break;
+            case ExampleEvent::BROKEN:
+            case ExampleEvent::FAILED:
+                $message = $this->getResultData($event, $event->getResult());
+                $result = sprintf(self::NOT_OK, $this->examplesCount) . $desc . "\n" . $message;
+                break;
+        }
+
+        $this->getIO()->writeln($result);
+    }
+
+    /**
+     * @param SuiteEvent $event
+     */
+    public function afterSuite(SuiteEvent $event)
+    {
+        $this->getIO()->writeln(sprintf(
+            self::PLAN,
+            $this->getStatisticsCollector()->getEventsCount()
+        ));
+    }
+
+    /**
+     * Format message as two-space indented YAML when needed outside of a
+     * SKIP or TODO directive.
+     *
+     * @param ExampleEvent $event
+     * @param int $result
+     * @return string
+     */
+    private function getResultData(ExampleEvent $event, $result = null)
+    {
+        if (null === $result) {
+            return $this->stripNewlines($event->getException()->getMessage());
+        }
+        $message = $event->getException()->getMessage();
+        switch ($result) {
+            case ExampleEvent::PENDING:
+                $message = sprintf(self::TODO, $this->stripNewlines($message))
+                    . $this->indent(sprintf(self::SEVERITY, Yaml::dump('todo')));
```

Comment:
Ah yes. `Yaml\Dumper::dump` is correctly hinted as `mixed` but the facade in `Yaml::dump` is `array` (and the parameter is called `$array`). My main concern was whether this might change but I guess not.